### PR TITLE
PLT-3547 fixing caching issue with direct profiles

### PR DESCRIPTION
**Pull request by coreyhulen**
#### Summary
fixes caching issue with direct profiles

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-3547

#### Checklist
- [ ] Added or updated unit tests (required for all new features)
- [ ] Added API documentation (required for all new APIs)
- [ ] Has driver changes that have been merged and package.json updated
- [ ] Has enterprise changes (please link)
- [ ] Has UI changes
- [ ] Includes text changes and localization files updated
- [ ] Touches critical sections of the codebase (auth, upgrade, etc.)

### CODE REVIEW

File: store/sql_user_store.go [hunk 1]
```diff
@@ -510,10 +512,12 @@ func (s SqlUserStore) GetEtagForDirectProfiles(userId string) StoreChannel {
...
         `, map[string]interface{}{"UserId": userId})
-		if err != nil {
-			result.Data = fmt.Sprintf("%v.%v.%v.%v", model.CurrentVersion, model.GetMillis(), utils.Cfg.PrivacySettings.ShowFullName, utils.Cfg.PrivacySettings.ShowEmailAddress)
+
+		if err != nil || len(ids) == 0 {
+			result.Data = fmt.Sprintf("%v.%v.0.%v.%v", model.CurrentVersion, model.GetMillis(), utils.Cfg.PrivacySettings.ShowFullName, utils.Cfg.PrivacySettings.ShowEmailAddress)
 		} else {
-			result.Data = fmt.Sprintf("%v.%v.%v.%v", model.CurrentVersion, updateAt, utils.Cfg.PrivacySettings.ShowFullName, utils.Cfg.PrivacySettings.ShowEmailAddress)
+			allIds := strings.Join(ids, "")
+			result.Data = fmt.Sprintf("%v.%v.%v.%v.%v", model.CurrentVersion, md5.Sum([]byte(allIds)), len(ids), utils.Cfg.PrivacySettings.ShowFullName, utils.Cfg.PrivacySettings.ShowEmailAddress)
```
Comment (jwilander) [inline store/sql_user_store.go]: Won't `allIds` always just contain a single id since the SQL query has `LIMIT 1`?

Comment (coreyhulen) [inline store/sql_user_store.go]: ugg...that would have been a nasty/similar bug to find.

Comment (coreyhulen) [inline store/sql_user_store.go]: @jwilander ready for re-review

**Comment by jwilander** [comment]
+1

**Comment by enahum** [comment]
+1

**Pull request merged**